core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/jupyter/JupyterHtmlRenderer.kt

```diff
@@ -60,21 +56,18 @@ internal inline fun <reified T : Any> JupyterHtmlRenderer.render(
     val staticHtml = df.toStaticHtml(reifiedDisplayConfiguration, DefaultCellRenderer).toJupyterHtmlData()
 
     if (notebook.kernelVersion >= KotlinKernelVersion.from(MIN_KERNEL_VERSION_FOR_NEW_TABLES_UI)!!) {
-        val jsonEncodedDf = json {
-            obj(
-                "nrow" to df.size.nrow,
-                "ncol" to df.size.ncol,
-                "columns" to df.columnNames(),
-                "kotlin_dataframe" to encodeFrame(df.rows().take(limit).toDataFrame()),
-            )
-        }.toJsonString()
-        notebook.renderAsIFrameAsNeeded(html, staticHtml, jsonEncodedDf)
+        val jsonEncodedDf = df.toJsonWithMetadata(limit, reifiedDisplayConfiguration.rowsLimit)
```

Comment:
Let's discuss this. I'd say default row limit that we used to have for HTML table might be too small in some situations. For example, imagine groupBy with 10 rows and 1000 rows in each group. I'd expect that it will still work very fast. That's why i'd prefer not to have any pagination in this situation at all.

Reply:
It is problematic. When loading a dataset with numerous rows within nested frames, such as ten frames with 1000 rows each. This can result in a large JSON file when serialized. It has a multiplicative effect when the number of cells increases.  It is even quite easy making it impossible to open in a Kotlin Notebook due to the default limit on JSON size during parsing. This limit is shared with the Python notebook plugin.

In my experience, pagination improves responsiveness and does not make it slower. You have the option to manually adjust the page size to view the entire nested table or utilize the “Go to Row” action. 

Another possible option is to include a function in KotlinNotebookPluginUtils that can modify or disable pagination for specific use cases.

Reply:
I see. Ok, i can agree with that. No need to change anything then